### PR TITLE
ref(interface-breadcrumbs): Make it responsive - INGEST-306

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x] Converted classes to function
- [x] Simplified code
- [x] Fixed minor issue in filter

Before:

image

After:

image

- [x] Updated minor logic in the tooltip component (if title is empty, it should return the children)
- [x] Make breadcrumb responsive 

**Previews**

large device:

image

image

smaller device:

image

closes: INGEST-306

### CODE REVIEW

**Comment by Jesse-Box** [comment]
Looks nice @priscilawebdev! Is it just me or do the icons look off-centre? The spanner icon looks the most off-centre but I couldn't tell you if it's just the icon or frontend trickery is needed.

**Comment by Jesse-Box** [comment]
🚢 it!

**Review by Jesse-Box — APPROVED** [review]

File: static/app/components/events/interfaces/breadcrumbs/breadcrumb/index.tsx [hunk 1]
```diff
@@ -1,79 +1,132 @@
...
+
+      /* Type */
+      :nth-child(5n-4) {
+        grid-row: 1/-1;
+        padding-right: 0;
+        padding-left: 0;
+        margin-left: ${space(2)};
+      }
+
+      /* Data */
```
Comment (matejminar) [inline static/app/components/events/interfaces/breadcrumbs/breadcrumb/index.tsx:91]: I appreciate these comments 👍

File: static/app/components/events/interfaces/breadcrumbs/breadcrumb/level.tsx [hunk 2]
```diff
@@ -14,35 +15,44 @@ const Level = memo(function Level({level, searchTerm = ''}: Props) {
...
+        <StyledTag>
+          <Highlight text={searchTerm}>{level || t('Undefined')}</Highlight>
+        </StyledTag>
       );
   }
 });
 
 export default Level;
+
+const StyledTag = styled(Tag)`
```
Comment (matejminar) [inline static/app/components/events/interfaces/breadcrumbs/breadcrumb/level.tsx:51]: How about `LevelTag` to be a bit more descriptive?

**Review by matejminar — COMMENTED** [review]
It's a bit hard to review this PR as it's tackling multiple things at once but I played with it locally and things are looking good.

One thing I've noticed previously is this jumping that's happening when scrolling breadcrumbs. I was confused and disoriented by this couple of times. I assume it's because of the virtualized list, can we tweak the styles so that it's less noticeable?

https://user-images.githubusercontent.com/9060071/134005349-6ca40dcb-d431-4b40-8f2b-ac614f42bfd1.mp4

**Comment by priscilawebdev** [comment]
thanks for the feedback @matejminar. I believe this is happening because of the virtualized list and can check if we can somehow improve this experience, but I would prefer to do it in a task/separate PR if that is fine 

Sorry for the unrelated changes. I saw some small issues working on the breadcrumbs and thought it would be fine to ship the fixes in this PR

**Review by matejminar — APPROVED** [review]

**Pull request merged**